--- Makefile	2017-03-11 11:36:14.104436756 +0200
+++ Makefile.patched.ra2d2	2017-03-11 11:41:29.217438271 +0200
@@ -155,6 +155,15 @@
 PROGRAMS += mod_cnc
 mod_cnc: $(mod_cnc_TARGET)
 
+# Red Alert 2
+mod_ra2_SRCS := $(shell find OpenRA.Mods.RA2/ -iname '*.cs')
+mod_ra2_TARGET = mods/ra2/OpenRA.Mods.RA2.dll
+mod_ra2_KIND = library
+mod_ra2_DEPS = $(STD_MOD_DEPS) $(mod_common_TARGET)
+mod_ra2_LIBS = $(COMMON_LIBS) $(STD_MOD_LIBS) $(mod_common_TARGET)
+PROGRAMS += mod_ra2
+mod_ra2: $(mod_ra2_TARGET)
+
 # Dune 2000
 mod_d2k_SRCS := $(shell find OpenRA.Mods.D2k/ -iname '*.cs')
 mod_d2k_TARGET = mods/d2k/OpenRA.Mods.D2k.dll
@@ -193,6 +202,9 @@
 	@echo "Checking for code style violations in OpenRA.Mods.D2k..."
 	@mono --debug OpenRA.Utility.exe ra --check-code-style OpenRA.Mods.D2k
 	@echo
+	@echo "Checking for code style violations in OpenRA.Mods.RA2..."
+	@mono --debug OpenRA.Utility.exe ra --check-code-style OpenRA.Mods.RA2
+	@echo
 	@echo "Checking for code style violations in OpenRA.Utility..."
 	@mono --debug OpenRA.Utility.exe ra --check-code-style OpenRA.Utility
 	@echo
@@ -226,6 +238,9 @@
 	@echo "Testing Tiberian Sun mod MiniYAML..."
 	@mono --debug OpenRA.Utility.exe ts --check-yaml
 	@echo
+	@echo "Testing Red Alert 2 mod MiniYAML..."
+	@mono --debug OpenRA.Utility.exe ra2 --check-yaml
+	@echo
 	@echo "Testing Dune 2000 mod MiniYAML..."
 	@mono --debug OpenRA.Utility.exe d2k --check-yaml
 	@echo
@@ -301,7 +316,7 @@
 
 package: all-dependencies core tools docs version
 
-mods: mod_common mod_cnc mod_d2k
+mods: mod_common mod_cnc mod_d2k mod_ra2
 
 all: dependencies core tools
 
@@ -368,6 +383,8 @@
 	@$(CP_R) mods/cnc "$(DATA_INSTALL_DIR)/mods/"
 	@$(CP_R) mods/ra "$(DATA_INSTALL_DIR)/mods/"
 	@$(CP_R) mods/d2k "$(DATA_INSTALL_DIR)/mods/"
+	@$(CP_R) mods/ra2 "$(DATA_INSTALL_DIR)/mods/"
+	@$(INSTALL_PROGRAM) $(mod_ra2_TARGET) "$(DATA_INSTALL_DIR)/mods/ra2"
 	@$(INSTALL_PROGRAM) $(mod_d2k_TARGET) "$(DATA_INSTALL_DIR)/mods/d2k"
 	@$(CP_R) mods/modchooser "$(DATA_INSTALL_DIR)/mods/"
 
