--- a/make.ps1
+++ b/make.ps1
@@ -108,7 +108,7 @@
 	
 	if ($version -ne $null)
 	{
-		$mods = @("mods/ra/mod.yaml", "mods/cnc/mod.yaml", "mods/d2k/mod.yaml", "mods/ts/mod.yaml", "mods/modchooser/mod.yaml", "mods/all/mod.yaml")
+		$mods = @("mods/ra/mod.yaml", "mods/cnc/mod.yaml", "mods/d2k/mod.yaml", "mods/ts/mod.yaml", "mods/ra2/mod.yaml", "mods/modchooser/mod.yaml", "mods/all/mod.yaml")
 		foreach ($mod in $mods)
 		{
 			$replacement = (gc $mod) -Replace "Version:.*", ("Version: {0}" -f $version)
@@ -136,6 +136,8 @@
 		echo "Testing mods..."
 		echo "Testing Tiberian Sun mod MiniYAML..."
 		./OpenRA.Utility.exe ts --check-yaml
+		echo "Testing Red Alert 2 mod MiniYAML..."
+		./OpenRA.Utility.exe ra2 --check-yaml
 		echo "Testing Dune 2000 mod MiniYAML..."
 		./OpenRA.Utility.exe d2k --check-yaml
 		echo "Testing Tiberian Dawn mod MiniYAML..."
@@ -170,6 +172,8 @@
 		./OpenRA.Utility.exe cnc --check-code-style OpenRA.Mods.D2k
 		echo "Checking for code style violations in OpenRA.Mods.TS..."
 		./OpenRA.Utility.exe cnc --check-code-style OpenRA.Mods.TS
+		echo "Checking for code style violations in OpenRA.Mods.RA2..."
+		./OpenRA.Utility.exe cnc --check-code-style OpenRA.Mods.RA2
 		echo "Checking for code style violations in OpenRA.Utility..."
 		./OpenRA.Utility.exe cnc --check-code-style OpenRA.Utility
 		echo "Checking for code style violations in OpenRA.Test..."
