--- a/make.ps1
+++ b/make.ps1
@@ -111,6 +111,8 @@
 		echo "Testing mods..."
 		echo "Testing Tiberian Sun mod MiniYAML..."
 		./OpenRA.Utility.exe ts --check-yaml
+		echo "Testing Red Alert 2 mod MiniYAML..."
+		./OpenRA.Utility.exe ra2 --check-yaml
 		echo "Testing Dune 2000 mod MiniYAML..."
 		./OpenRA.Utility.exe d2k --check-yaml
 		echo "Testing Tiberian Dawn mod MiniYAML..."
@@ -145,6 +147,8 @@
 		./OpenRA.Utility.exe cnc --check-code-style OpenRA.Mods.D2k
 		echo "Checking for code style violations in OpenRA.Mods.TS..."
 		./OpenRA.Utility.exe cnc --check-code-style OpenRA.Mods.TS
+		echo "Checking for code style violations in OpenRA.Mods.RA2..."
+		./OpenRA.Utility.exe cnc --check-code-style OpenRA.Mods.RA2
 		echo "Checking for code style violations in OpenRA.Utility..."
 		./OpenRA.Utility.exe cnc --check-code-style OpenRA.Utility
 		echo "Checking for code style violations in OpenRA.Test..."
